website/source/docs/connect/mesh_gateway.html.md

```diff
@@ -0,0 +1,169 @@
+---
+layout: "docs"
+page_title: "Connect - Mesh Gateways"
+sidebar_current: "docs-connect-meshgateways"
+description: |-
+  A Mesh Gateway enables better routing of a Connect service's data to upstreams in other datacenters. This section details how to use Envoy and describes how you can plug in a gateway of your choice.
+---
+
+# Mesh Gateways
+
+Mesh gateways enable routing of Connect traffic between different Consul datacenters. Those datacenters
+can reside in different clouds or runtime environments where general interconnectivity between all services
+in all datacenters isn't feasible. These gateways operate by sniffing the SNI header out of the Connect session
+and then route the connection to the appropriate destination based on the server name requested. The data
+within the Connect session is not decrypted by the Gateway.
+
+![Mesh Gateway Architecture](/assets/images/mesh-gateways.png)
```

Comment:
Not worth holding the docs for, but @opihana is there any chance that we have a prettier version of this graphic?

Reply:
I curled the one from our roadmap preview. It must exist somewhere in a nicer form.